src/Temporalio/Workflows/WorkflowQueryDefinition.cs

```diff
@@ -10,11 +11,31 @@ namespace Temporalio.Workflows
     /// </summary>
     public class WorkflowQueryDefinition
     {
+        /// <summary>
+        /// All known reserved query handler prefixes.
+        /// </summary>
+        internal static readonly string[] ReservedQueryHandlerPrefixes =
+        {
+            TemporalRuntime.ReservedNamePrefix,
+            "__stack_trace",
+            "__enhanced_stack_trace",
+        };
+
         private static readonly ConcurrentDictionary<MethodInfo, WorkflowQueryDefinition> MethodDefinitions = new();
         private static readonly ConcurrentDictionary<PropertyInfo, WorkflowQueryDefinition> PropertyDefinitions = new();
 
-        private WorkflowQueryDefinition(string? name, string? description, MethodInfo? method, Delegate? del)
+        private WorkflowQueryDefinition(string? name, string? description, MethodInfo? method, Delegate? del, bool bypassReserved = false)
         {
+            if (!bypassReserved && name != null)
```

Comment:
If you want to save lines, add `using System.Linq;` import and then:

```suggestion
            if (!bypassReserved && name != null && ReservedQueryHandlerPrefixes.Any(p => name.StartsWith(p))
```

Reply:
Don't save all that much if you want to keep the prefix name in the error which I do, but, changed.